src/main/java/org/opensearch/performanceanalyzer/rca/net/handler/PublishRequestHandler.java

```diff
@@ -74,12 +74,15 @@ public PublishRequestHandler(
 
     public void terminateUpstreamConnections() {
         for (final StreamObserver<PublishResponse> responseStream : upstreamResponseStreamList) {
+            /* TODO: We need to check somehow to see if stream was already completed before calling onNext.
```

Comment:
Can we create an issue for this?

Reply:
Had already created one - https://github.com/opensearch-project/performance-analyzer-rca/issues/131. And raised PR for that. Will remove this comment eventually.